docs/source/data/subscriptions.mdx

```diff
@@ -354,84 +440,42 @@ Context object for the WebSocket connection. This is _not_ the `context` object
+On the client, `SubscriptionsClient` supports adding token information to `connectionParams` ([example](https://www.apollographql.com/docs/react/advanced/subscriptions/#authentication-over-websocket)) that will be sent with the first WebSocket message. On the server, all GraphQL subscriptions are delayed until the connection has been fully authenticated and the `onConnect` callback returns a truthy value.
 
 The `connectionParams` argument in the `onConnect` callback contains the information passed by the client and can be used to validate user credentials.
 The GraphQL context can also be extended with the authenticated user data to enable fine grain authorization.
 
 ```js
-const { ApolloServer } = require('apollo-server');
-const { resolvers, typeDefs } = require('./schema');
-
-const validateToken = authToken => {
+async function validateToken(authToken) {
   // ... validate token and return a Promise, rejects in case of an error
 };
 
-const findUser = authToken => {
-  return tokenValidationResult => {
+function findUser(authToken) {
+  return async function (tokenValidationResult) {
     // ... finds user by auth token and return a Promise, rejects in case of an error
   };
 };
 
-const server = new ApolloServer({
-  typeDefs,
-  resolvers,
-  subscriptions: {
-    onConnect: (connectionParams, webSocket) => {
-      if (connectionParams.authToken) {
-        return validateToken(connectionParams.authToken)
-          .then(findUser(connectionParams.authToken))
-          .then(user => {
-            return {
-              currentUser: user,
-            };
-          });
+SubscriptionServer.create({
+  schema,
+  execute,
+  subscribe,
+  async onConnect(connectionParams, webSocket) {
+    if (connectionParams.authToken) {
+      try {
+        const validationResult = await validateToken(connectionParams.authToken);
+        const currentUser = await findUser(connectionParams.authToken)(validation);
+        return { currentUser };
       }
-
-      throw new Error('Missing auth token!');
-    },
+    }
+    throw new Error('Missing auth token!');
   },
 });
-
-server.listen().then(({ url, subscriptionsUrl }) => {
-  console.log(`🚀 Server ready at ${url}`);
-  console.log(`🚀 Subscriptions ready at ${subscriptionsUrl}`);
-});
 ```
 
-The example above validates the user's token that is sent with the first initialization message on the transport, then it looks up the user and returns the user object as a Promise. The user object found will be available as `context.currentUser` in your GraphQL resolvers.
-
-In case of an authentication error, the Promise will be rejected, which prevents the client's connection.
-
+The example above validates the user's token that is sent with the first initialization message on the transport, then it looks up the user and returns the user object as a `Promise`. The user object found will be available as `context.currentUser` in your GraphQL resolvers.
```

Comment:
Now that they're async functions we probably don't have to talk about "as a Promise" in this or the next paragraph

Reply:
Interesting. I don't really differentiate or think that differently between using `async/await` and `Promise` objects. Like when I look at an `async` function I just think of it as a function returning a `Promise`. I think it's not not helpful to leave it as is, plus immediately below it talks about if the `Promise` is rejected. Thoughts?

Reply:
Well, my suggestion was to change the next part too, but sure.

I think you can think about async functions either as functions returning Promises, or just as functions that get to call await.  Whether throwing in an async function is "throwing" or "rejecting the returned Promise" depends on how you think about it.  I think probably it's a simpler but less accurate mental model to think about returning/throwing. If you think the current text is clearer to you then let's leave it.